src/software/ai/hl/stp/tactic/goalie/goalie_tactic.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "software/ai/hl/stp/tactic/goalie/goalie_fsm.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+
+/**
+ * This tactic is used to defend the ball from going into the goal. The tactic
+ * is assigned to the robot that is selected using a DynamicParameter, and stays
+ * that way throughout all the plays that require a goalie.
+ *
+ * If the ball is moving faster than a threshold towards the net, moves to intercept
+ * the ball. If not, returns intents that position the robot in a cone between the ball
+ * and the two goal posts, in such a way that the robot would have to move a minimal
+ * distance either way to intercept a potential straight shot into the net.
+ *
+ */
+class GoalieTactic : public Tactic
+{
+   public:
+    /**
+     * Creates a new GoalieTactic
+     *
+     * @param goalie_tactic_config The config to fetch parameters from
+     */
+    explicit GoalieTactic(std::shared_ptr<const GoalieTacticConfig> goalie_tactic_config);
+
+    GoalieTactic() = delete;
+
+    void updateWorldParams(const World &world) override;
+
+    /**
+     * Updates the params assuming that the max allowed speed mode is the physical limits
+     * @param updated_goalie_tactic_config the updated goalie tactic config
+     */
+    void updateControlParams(
+        std::shared_ptr<const GoalieTacticConfig> updated_goalie_tactic_config);
```

Comment:
I don't think this would ever require updating, i.e. if config changes, then the user should construct a new goalie, so it can be removed

Reply:
Sounds good, to clarify do you mean remove the function altogether?

Reply:
yes

Reply:
removed!